main.go

```diff
@@ -193,12 +287,34 @@ func StdoutMatches(regex string) bool {
 
 // StderrMatches compares a regex to the stderr produced by the command.
 func (c *Cmd) StderrMatches(regex string) bool {
-	c.validate()
+	c.validateHasStarted()
 	re := regexp.MustCompile(regex)
-	return re.MatchString(c.Stderr())
+	return retryStringTest(func(got, want string) bool {
+		return re.MatchString(got)
+	}, c.stderr, regex)
 }
 
 // StderrMatches compares a regex to the stderr produced by the command.
 func StderrMatches(regex string) bool {
 	return pkgCmd.StderrMatches(regex)
 }
+
+// retryStringTest takes in a testFunc and will test output for the expected string until either it
+// finds the expected string or times out (default 1 second)
+func retryStringTest(testFunc func(string, string) bool, output *output, expected string) bool {
```

Comment:
What is the use case for this method?

Reply:
As mentioned in the comment on the mutex I added a retry mechanism in the `*Contains` and `*Matches` methods to prevent race conditions as the output sometimes seemed to be delayed. It also expands the usecase to long running commands/processes that are delayed in producing the desired output. The `testFunc` can be something like `strings.Contains`, and will be executed with the content of the `output` and `expected` string every 100ms until either a match is found or it reaches the timeout. 

At the moment the timeout  (and interval) is hardcoded but this should be configurable I suppose so it can be tailored to specific usecases